### PR TITLE
[feature] Nereids code base

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

Issue Number: #6483

## Problem Summary:

Nereids(new optimizer) code base

Nereids is new query planner for Doris. It include three main parts: parser, analyzer and optimizer.
The parser, generated by ANTLR4, transforms SQL into a logical plan with a tree structure. Analysis and optimization are performed on the logical plan of the tree structure. Each transformation is defined as a rule. The rule is applied to the logical plan using pattern matching. The implementation of the optimizer follows the approach in the Cascades paper.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (Yes)
5. Are there any changes that cannot be rolled back: (No)

### CODE REVIEW

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/common/util/VectorizedUtil.java [hunk 1]
```diff
@@ -37,7 +37,10 @@ public static boolean isVectorized() {
         if (stmtExecutor == null) {
             return connectContext.getSessionVariable().enableVectorizedEngine();
         }
-        Analyzer analyzer = stmtExecutor.getAnalyzer();
+        Analyzer analyzer = null;
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/VectorizedUtil.java:40]: Your pr should not bring this change, is it caused by no rebase master?

**Comment by EmmyMiao87** [comment]
Notice the antlr license in dist/NOTICE-dist.txt

File: fe/fe-core/src/main/java/org/apache/doris/common/util/VectorizedUtil.java (hunk 1 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/common/util/VectorizedUtil.java:40]: revert it~

**Comment by morrySnow** [comment]
> Notice the antlr license in dist/NOTICE-dist.txt

done

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**